### DISCUSSION TITLE
Smart Filters containing Multiple Items

### DISCUSSION THREAD

**Discussion by mrittlinger** [provectus/kafka-ui#3965] [Q&A]
I'm having some trouble getting the smart filters to work with more complex constructs. I am using 0.7.

My value is json format and looks like this:
```
{
	"IsError": false,
	"Message": "Unauthorized",
	"ResponseContent": "",
	"RequestContent": "",
	"EventType": "fill",
	"SystemId": "5670",
	"Payload": "redacted"
}
```

I can filter by a single value. Both of these filters work fine individually:

`value.SystemId == "5670"`

`value.Message.contains("Unauthorized")`

If I try to combine them the data is not filtered (as if there is no filter). Clicking the "Submit" button returns immediately. I have tried these variations:

`value.SystemId == "5670" && value.Message ==~ "Unauthorized*"`

`value.SystemId == "5670" && value.Message.contains("Unauthorized")`
```
def systemid = value.SystemId
def msg = value.Message
systemid == "5670" && msg == "Unauthorized"
```
```
def systemid = value.SystemId
def msg = value.Message
systemid == "5670" && msg.contains("Unauthorized")
```

Similarly, I have also attempted to use the timestampMs field to filter by date range. 
Groovy attempt -
```
import groovy.time.*

def startString = "2023-06-20T05:00:00.0Z"
def endString = "2023-06-20T06:00:00.0Z"

def start = Date.parse("yyyy-MM-dd'T'HH:mm:ss",startString).getTime()
def end = Date.parse("yyyy-MM-dd'T'HH:mm:ss",endString).getTime()

timestampMs >= start && timestampMs <= end
```
Java attempt -
```
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.time.ZoneId;

var startString = "2023-06-20T05:00:00Z";
var endString = "2023-06-20T12:00:00Z";

var formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'");

var start = LocalDateTime.parse(startString, formatter).atZone(ZoneId.systemDefault()).toInstant().toEpochMilli();
var end = LocalDateTime.parse(endString, formatter).atZone(ZoneId.systemDefault()).toInstant().toEpochMilli();

timestampMs >= start && timestampMs <= end;
```

Any advice on how to filter by multiple fields and date range would be greatly appreciated.

**Comment by iliax** [comment]
Hello @mrittlinger ! Thanks for raising it. 
I just checked your scenario on v0.7.0 and I can't reproduce it. 
Please check the screenshots - maybe I am doing smth wrong. 

Messages in topic:
<img width="687" alt="Screenshot 2023-07-12 at 11 44 37" src="https://github.com/provectus/kafka-ui/assets/702205/7a5df130-60d9-408f-9037-fdc2019fecda">

Adding filter:
<img width="498" alt="Screenshot 2023-07-12 at 11 44 55" src="https://github.com/provectus/kafka-ui/assets/702205/eb3575a3-86aa-4e97-a9b5-70a6bcb63f23">

Result:
<img width="1361" alt="Screenshot 2023-07-12 at 11 45 05" src="https://github.com/provectus/kafka-ui/assets/702205/ce42b86d-b0e5-42b9-afe5-494c990a4381">


-----------
-----------


Here are some questions:
1. what browser do you use? 
2. are there any errors in kui logs?
3. are there errors showing in the right corner of messages view (see screenshot) ?
<img width="470" alt="Screenshot 2023-07-12 at 11 53 12" src="https://github.com/provectus/kafka-ui/assets/702205/46a11076-1524-4eba-9c55-2daaf9614207">

**Reply by ritchApps** [reply]
is the browser important?
what If I don't see any errors?
I'm not getting my messages filtered 
![Screenshot 2024-06-04 at 12 59 09 PM](https://github.com/provectus/kafka-ui/assets/2043934/7cd44f40-b82d-4100-9ebf-e25d9d219b7c)

**Reply by Haarolean** [reply]
@ritchApps https://github.com/provectus/kafka-ui/issues/4437#issuecomment-2077685765

**Comment by bidyut90** [comment]
Hello team, I'm having trouble getting the date rage. Could someone please assist me?

**Reply by Haarolean** [reply]
https://github.com/provectus/kafka-ui/issues/4437#issuecomment-2077685765

**Comment by benny-yamagata** [comment]
Just FYI, getting the same issue. 
0.7.2 - Smart filters not working at all. 
Getting 400 bad request in the console when I click submit.

**Reply by Haarolean** [reply]
https://github.com/provectus/kafka-ui/issues/4437#issuecomment-2077685765
